core/src/main/java/org/apache/carbondata/scan/executor/impl/AbstractQueryExecutor.java

```diff
@@ -470,7 +471,12 @@ private AggregatorInfo getAggregatorInfoForBlock(QueryModel queryModel,
    */
   @Override public void finish() throws QueryExecutionException {
     if (null != queryProperties.executorService) {
-      queryProperties.executorService.shutdownNow();
+      queryProperties.executorService.shutdown();
+      try {
+        queryProperties.executorService.awaitTermination(1, TimeUnit.HOURS);
```

Comment:
Its a max waiting time for completing the scanning if processing is done, then it will come out automatically, so i think it is ok